File: apple/Sources/ReactNativeWatchOS/Modifiers/Padding.swift
import SwiftUI

/// SwiftUI `.padding(_:)`. Per-edge values accumulate on top of `all` /
/// `horizontal` / `vertical` shortcuts — `{horizontal: 8, top: 12}` ⇒
/// leading=trailing=8, top=12, bottom=0.
enum RNWPaddingModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("padding") { view, params, _ in
            var insets = EdgeInsets()
            if let all = params.double("all") {
                insets = EdgeInsets(
                    top: all, leading: all, bottom: all, trailing: all)
            }
            if let h = params.double("horizontal") {
                insets.leading = h
                insets.trailing = h
            }
            if let v = params.double("vertical") {
                insets.top = v
                insets.bottom = v
            }
            if let t  = params.double("top")      { insets.top = t }
            if let b  = params.double("bottom")   { insets.bottom = b }
            if let l  = params.double("leading")  { insets.leading = l }
            if let tr = params.double("trailing") { insets.trailing = tr }
            return AnyView(view.padding(insets))
        }
    }
}

File: src/modifiers/padding.ts
import { createModifier } from './createModifier';

export interface PaddingParams {
  /// Uniform padding on all four edges. Overridden by more specific keys.
  all?: number;
  /// Horizontal padding (leading + trailing).
  horizontal?: number;
  /// Vertical padding (top + bottom).
  vertical?: number;
  top?: number;
  bottom?: number;
  leading?: number;
  trailing?: number;
}

/// SwiftUI `.padding(_:)`. Pass no arg for system default padding (8pt on
/// watchOS); pass a number for uniform padding; pass an object for
/// per-edge values. Later keys override earlier ones.
export function padding(params?: PaddingParams | number) {
  if (typeof params === 'number') {
    return createModifier('padding', { all: params });
  }
  return createModifier('padding', params);
}
